MetricsUploader/MetricsUploader.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "MetricsUploader/MetricsUploader.h"
+
+#include <absl/memory/memory.h>
+
+#include <vector>
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_metrics_uploader {
+
+absl::Mutex MetricsUploader::init_mutex_;
+
+std::string MetricsUploader::GetErrorMessage(Result result) {
+  switch (result) {
+    case kMetricsUploaderServiceNotStarted:
+      return "Metrics uploader service was not started";
+    case kSdkConfigNotLoaded:
+      return "SDK config was not loaded";
+    case kCannotMarshalLogEvent:
+      return "Can't marshall log event";
+    case kCannotQueueLogEvent:
+      return "Can't queue log event";
+    case kClientNotInitialized:
+      return "Uploader client wasn't initialized. StartUploaderClient() should be called before "
+             "sending log events.";
+    case kCannotUnmarshalLogEvent:
+      return "Can't unmarshal OrbitLogEvent proto";
+    default:
+      return "No errors";
+  }
+}
+
+ErrorMessageOr<std::unique_ptr<MetricsUploader>> MetricsUploader::CreateMetricsUploader(
+    const std::string_view client_name) {
+  absl::MutexLock lock(&init_mutex_);
+  if (is_init_) {
+    return ErrorMessage("MetricsUploader is already created");
+  }
+  MetricsUploader* metrics_uploader = new MetricsUploader(client_name);
```

Comment:
You could use std::make_unique here instead of `new`.

Reply:
I'm not sure I can do it due to private constructor: https://abseil.io/tips/134

Reply:
Ah, yes you're right. In this case I'd would recommend to move absl::WrapUnique to line 43. This way you still have a unique_ptr the whole.

BTW, you can also write `const auto metrics_uploader = std::unique_ptr{new MetricsUploader(client_name)};`. AFAIK `WrapUnique` also just calls the constructor of unique_ptr.

Reply:
Done